mos-platform/common/include/string.h

```diff
@@ -32,6 +32,9 @@ size_t strspn(const char* string, const char* in);
 char* strpbrk(const char* string, const char* brk);
 char* strtok(char* string, const char* separators);
 
+char *strcat(char * __restrict__ s1, const char * __restrict__ s2);
```

Comment:
It's not used elsewhere, but perhaps Doxygen style comments could be added? Some IDE's can pick them up and display them as popup help.

Reply:
Perhaps copy in `.clang-format` and `.clang-tidy` from the llvm-mos rep? Styles vary throughout `string.h`, e.g. `char*` vs `char *`.

Reply:
It is used elsewhere in the repo - I've added them to my PC Engine and RPC8/e libraries. As such, there is precedent.

Reply:
Sure, I can get the llvm clang-format (I've think I have done it previously, but I get pristine or work on a different machine and it goes away). Why not add the format you desire to the tree so it (mostly) defaults?  I didn't feel comfortable reformatting the whole file (but I would love not having to think much about formatting - I need some kind of clang format to write code and show me where I need to balance braces 🙂).  I looked a bit into .clang-tidy, but not clear how to integrate into my git flow (I am not great with git at this point).

Reply:
`.clang-format` should probably be added and applied in a separate PR.

Reply:
Okay, I verified string.c did match up llvm clang-format (at least didn't change on reformat).  However, now I remember, I really dislike the llvm clang-format brace dangling on the end thing, so do I really have to use it on the X16 examples I create?😅  It wasn't mentioned on the other examples I previously submitted.  But, if that is policy, then so be it. I can understand keeping it all consistent and in lib code and files I didn't create I have been trying to be a "code chameleon" (after some previous newbie mistakes).🦎

Reply:
Also, I was not really aware of Doxygen use.  Cool, going forward I can look into decorating some things to make that (more) useful.

Reply:
I guess it's just a matter of taste and probably something Daniel should decide on at a project level. In any case, it can/should be changed at a later stage so no worries.

Reply:
I've been really lax on this, but now that it's a question, enforcing the llvm style is probably the most straightforward non-terrible thing we can do. It has machine support via clang-format, and it's relatively easy to git clang-format a commit before it goes in.

Agree that this should be separate; we should do a refactoring for style once and only once; otherwise we'll cut the git history more than is strictly necessary.